### PR TITLE
fix clippy warnings

### LINKED ISSUE #582: master build is broken by new lint rules introduced by rust 1.53
**Describe the bug**

master build is broken by new lint rules introduced by rust 1.53

[context][1]

**To Reproduce**

**Expected behavior**

**Additional context**

[1]: https://github.com/rust-lang/rust-clippy/blob/master/CHANGELOG.md#enhancements

### LINKED ISSUE #583: change all quotes to smart quotes within error messages
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

so far some of the test messages require escaping either single `'` or double `"` quotes. this is error prone and and sometimes introduces confusions.

**Describe the solution you'd like**

we can easily resolve this issue by replacing all current and future usage of quotes into smart quotes `‘` `’` and `“` `”`

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Jimexist**
# Which issue does this PR close?

Closes #583 
Closes #582

 # Rationale for this change

newest clippy will generate additional warnings on master

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/src/logical_plan/dfschema.rs [hunk 1]
```diff
@@ -248,12 +248,14 @@ where
 }
 
 impl ToDFSchema for Schema {
+    #[allow(clippy::wrong_self_convention)]
```
Comment (Jimexist) [inline datafusion/src/logical_plan/dfschema.rs:251]: maybe it's a better idea to remove https://github.com/apache/arrow-datafusion/pull/581/files#diff-54ef5cfee73dd6cdbd9186bbf5931f4e375ccf002f956d1fe8bf35deafaf56a6R244

**Comment by Jimexist** [comment]
Maybe for the moment it is wise to temporarily disable coverage task in CI?

**Comment by alamb** [comment]
Not sure what the coverage failure was: https://github.com/apache/arrow-datafusion/pull/581/checks?check_run_id=2858343109 

I restarted the tests

File: datafusion/src/logical_plan/dfschema.rs (hunk 1 above)
Comment (alamb) [reply datafusion/src/logical_plan/dfschema.rs:251]: I don't understand the suggestion

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -349,9 +349,7 @@ jobs:
           export ARROW_TEST_DATA=$(pwd)/testing/data
           export PARQUET_TEST_DATA=$(pwd)/parquet-testing/data
 
-          # 2020-11-15: There is a cargo-tarpaulin regression in 0.17.0
-          # see https://github.com/xd009642/tarpaulin/issues/618
-          cargo install --version 0.16.0 cargo-tarpaulin
+          cargo install --version 0.18.0-alpha3 cargo-tarpaulin
```
Comment (alamb) [inline .github/workflows/rust.yml:352]: I don't understand this change. It seems to cause an error in CI, right?: https://github.com/apache/arrow-datafusion/pull/581/checks?check_run_id=2858343109

Maybe we could put the old version of tarpaulin back?

**Review by alamb — APPROVED** [review]
Thank you @Jimexist

File: .github/workflows/rust.yml (hunk 2 above)
Comment (Jimexist) [reply .github/workflows/rust.yml:352]: It was failing either way so this line diff was my attempt to fix it in vain

**Comment by Jimexist** [comment]
> Not sure what the coverage failure was: https://github.com/apache/arrow-datafusion/pull/581/checks?check_run_id=2858343109 
> 
> 
> 
> I restarted the tests

I tried several times and the error persists - not sure why given the fixture is now raw string so escaping should be out of picture

File: .github/workflows/rust.yml (hunk 2 above)
Comment (alamb) [reply .github/workflows/rust.yml:352]: I see -- in that case it does seem like disabling code coverage until it is fixed is the right thing to do, and we can file an issue to re-enable it once whatever underlying issue is fixed

Comment (alamb) [reply .github/workflows/rust.yml:352]: Filed https://github.com/apache/arrow-datafusion/issues/590 to fix code coverage job. Will disable it via this PR

**Comment by alamb** [comment]
I pushed https://github.com/apache/arrow-datafusion/pull/581/commits/ac7808ba2dea0fde5a1dce280529b71f819e80bc to disable the coverage job. Hopefully the CI checks will then pass and we can get master back to green.

Thank you for the help @jim

**Comment by alamb** [comment]
Thanks again @Jimexist

**Pull request merged**